### PR TITLE
F 301 hourly conversion rate is not working

### LINKED ISSUE #301: Hourly conversion rate is not working
Develop cannot set `usdValue` of new donations. Maybe new update on hourly conversion rate be the cause.
giveth/giveth-dapp#1566
#287 
#298

### PR DESCRIPTION
**Pull request by mohammadranjbarz**
Fix #301 

We should use `rateEqSymbol` when it's exists, 
Add some tests for testing these changes

### CODE REVIEW

File: src/services/conversionRates/getConversionRatesService.js [hunk 1]
```diff
@@ -203,18 +203,36 @@ const getHourlyRateCoingecko = async (
...
+    tokenCompareHistoryResponse &&
+    tokenCompareHistoryResponse.Data &&
+    // when there is no result and token is invalid, the resp.Data is {} not an array
+    Array.isArray(tokenCompareHistoryResponse.Data) &&
+    tokenCompareHistoryResponse.Data.length > 0 &&
+    tokenCompareHistoryResponse.Data.sort((a, b) => {
+      return b.time - a.time;
+    })[0]
+  );
+};
```
Comment (mohammadranjbarz) [inline src/services/conversionRates/getConversionRatesService.js:218]: If you call cryptocompare https://min-api.cryptocompare.com/data/histohour?fsym=DAI&tsym=BTC&toTs=1607748079&extraParams=giveth&limit=1

you will see the response is like this :

```
{
    "Response": "Success",
    "Type": 100,
    "Aggregated": false,
    "TimeTo": 1607745600,
    "TimeFrom": 1607742000,
    "FirstValueInArray": true,
    "ConversionType": {
        "type": "invert_divide",
        "conversionSymbol": "ETH"
    },
    "Data": [
        {
            "time": 1607742000,
            "high": 0.00005494,
            "low": 0.00005477,
            "open": 0.00005481,
            "volumefrom": 41444.98,
            "volumeto": 2.274,
            "close": 0.0000549,
            "conversionType": "invert",
            "conversionSymbol": ""
        },
        {
            "time": 1607745600,
            "high": 0.00005473,
            "low": 0.00005441,
            "open": 0.00005495,
            "volumefrom": 13936.37,
            "volumeto": 0.762,
            "close": 0.00005468,
            "conversionType": "invert_divide",
            "conversionSymbol": "ETH"
        }
    ],
    "RateLimit": {},
    "HasWarning": false
}
```


@aminlatifi  and none of objects doesn't have field `time` equals to timestamp that we send (in this example :1607748079),
so finding object with` time` equals to `timestampMS` always return undefined, and for solving this problem I sort the array and return the newest one, 
this function have tests , you can check it out

**Comment by mohammadranjbarz** [comment]
And in develop I checkout to this branch and restart application,
you can check this donation that usd value has been set correctly:

https://feathers.develop.giveth.io/donations/5fd467a6ccd76a21b11f0a28
this donations is one of donations of below milestone that I donated after deploy:

https://develop.giveth.io/campaigns/5fc69bc0048ee557f352f525/milestones/5fc6ae19048ee557f352f558

**Comment by mohammadranjbarz** [comment]
@aminlatifi 
Now Mahdi test passed, and I added more test cases.
( The test didn't pass because if you send similar tokens you will get error, we should resolve similar token and dont request to cryptocompare)

**Comment by aminlatifi** [comment]
@mohammadranjbarz @mahdine Thanks for you effort.
I did some modification to support other token to PAN conversion.

**Pull request merged**